app/forms/state_file/nc_taxes_owed_form.rb

```diff
@@ -1,63 +1,49 @@
+    with_options unless: -> { payment_or_deposit_type == "mail" || !form_submitted_before_payment_deadline? } do
+      validate :withdrawal_date_is_not_today
+      validate :withdrawal_date_is_not_on_a_weekend
+      validate :withdrawal_date_is_not_a_federal_holiday
+      validate :withdrawal_date_is_at_least_two_business_days_in_the_future_if_after_5pm
     end
 
+    private
 
-    def date_electronic_withdrawal_nc_validations
-      five_pm_et = app_time_eastern.change(hour: 17, min: 0, sec: 0)
-      if (app_time_eastern >= five_pm_et) && less_than_two_business_days_in_future?
-        errors.add(:date_electronic_withdrawal, I18n.t("errors.attributes.nc_withdrawal_date.post_five_pm"))
-      end
-      if date_electronic_withdrawal <= app_time_eastern.to_date
+    def withdrawal_date_is_not_today
+      unless date_electronic_withdrawal.after?(@form_submitted_time.to_date)
         errors.add(:date_electronic_withdrawal, I18n.t("errors.attributes.nc_withdrawal_date.past"))
       end
+    end
+
+    def withdrawal_date_is_not_on_a_weekend
       if date_electronic_withdrawal.saturday? || date_electronic_withdrawal.sunday?
         errors.add(:date_electronic_withdrawal, I18n.t("errors.attributes.nc_withdrawal_date.weekend"))
       end
+    end
+
+    def withdrawal_date_is_not_a_federal_holiday
       if Holidays.on(date_electronic_withdrawal, :us, :federalreservebanks, :observed).any?
         errors.add(:date_electronic_withdrawal, I18n.t("errors.attributes.nc_withdrawal_date.holiday"))
       end
     end
 
-    private
-
-    def app_time_eastern
-      app_time.in_time_zone('Eastern Time (US & Canada)')
-    end
-
-    def add_business_days(num_days)
-      current_date = app_time_eastern.to_date
-      while num_days > 0
-        current_date += 1.day
-        # Check if the current date is a business day (Mon-Fri)
-        if current_date.wday.between?(1, 5)  # wday 1-5 are Monday to Friday
-          num_days -= 1.day
-        end
+    def withdrawal_date_is_at_least_two_business_days_in_the_future_if_after_5pm
+      # From the ticket (FYST-1061):
+      # if you submit your bank draft payment after 5:00 pm EST, the earliest draft date available will be two business days in the future
+      after_5pm = @form_submitted_time.hour >= 17
+      two_business_days_away = add_business_days_to_date(@form_submitted_time.to_date, 2)
+      if after_5pm && !date_electronic_withdrawal.to_date.after?(two_business_days_away)
+        errors.add(:date_electronic_withdrawal, I18n.t("errors.attributes.nc_withdrawal_date.post_five_pm"))
       end
-      current_date
     end
 
-    def less_than_two_business_days_in_future?
-      two_business_days_later = add_business_days(2)
-      date_electronic_withdrawal <= two_business_days_later
+    def add_business_days_to_date(date, num_days)
+      while num_days.positive?
+        date += 1.day
+        num_days -= 1 if date.wday.between?(1, 5)
```

Comment:
i kind of like the comment:
`# Check if the current date is a business day (Mon-Fri)` here for obscure `wday` method